### ISSUE TITLE
x86 MID hardware failures catchall

### ISSUE THREAD

**Issue by bobbybudnick** [bobbybudnick/SFS-Mobile-Internet-Device#46]
Description:  

Typical problems
screen backlight flickers
screen display cuts in and out
frequent issues with touch input
cellular issues with good signal       
usb emi errors in kernel log

Existing test configuration
4.3" screen and cellular connected to hub and hub 5v input is switched
this caused the usual problems  
problem stopped when screen connected to external power
problem got much worse after plugging in another hub and keyboard

Explanation:  

The first series of tests and brainstorming was put into action for this problem.

idea 1
switch hub at output instead of input and put type a power in parallel
too complicated and still uses up 2 type a connectors
not implemented

idea 2
screen gets type a and cellular gets type a with common 5v switch
implemented with failure
exposes much of the data lines to interference
less simple in practice than theory to create   
high potential for loose and shorted connections
results in damage and sacrifice of wiring
a mess to integrate
was somewhat unreliable with hub before but now far worse
totally unusable this way to a maximum degree
yet we will see that in some configurations this works ok

idea 3
direct wire screen usb to hub output
probably not going to be satisfactory
kind of a band aid fix
also breaks shielding in hub wiring
not implemented

None of those ideas were successful so we move on to an expanded range of troubleshooting possibilities.

strategy 1
improve loose and broken wiring at hub switch
largely fixed but there is not much room to work

strategy 2
use of the hub in general
the hub is actually shielded from emi
it may have power backfeed features that also help
only the hub power lines need to be modifed and data does not
shielding is left largely intact
only modification is on the hub input side and no others
wiring is kept simpler
shielded cables can be used downstream
being usb 2 may help because cellular tries to connect at usb 3 speeds otherwise

strategy 3
use of second type a connector positive wire for hub switch
takes up a second port but leaves one free
a booster if screen dimming is still a problem

strategy 4
use of external power microusb connected to second type a and hub switch
hopefully touch will still work this way - there is a concern about external power port
another booster if screen dimming is still a problem

strategy 5
use of 5" screen
4.3" screen seems more susceptible to interference
constant errors may take out all usb devices
5" naturally expands and reduces emi zone
test existing model with 5" screen

keyboard wiring
if strategy 1/2 is a success then use unmodified microusb in second port
if strategy 3 is a succcess then the third port would be needed
if strategy 4 is a success then custom wiring can be used

The results for each individual strategy will be discussed here.

strategy 1/2
screen is still in brownout flickering state
already very much more stable as far as dmesg errors and touchscreen failures
some screen disconnections after a while

strategy 3
some disconnect and flickering and usb emi errors still
also total system freeze
is ground attached to second type a port??? - ground was not attached

strategy 3 with second type a ground connected
control test with second type a not connected
failures begin when we went to open the web browser and load up 5v with cellular
----- set max efficiency 7w in bios - turbo mode keep on - force 5w in software
repeat test again after bios change
strategy 3 with second type a ground connected
control test with second type a not connected
flickering and standard failures

strategy 3 with second type a ground connected
non control test with second type a connected
works noticeably better than ever before

strategy 4 - not tested yet
external power connected - may not initialize touch correctly

strategy 5 - not tested yet
admit defeat and go with the 5" screen

extended strategy - not tested yet
look heavily into bios options and consider emi from alpha model is just too great
very telling that delta/5" is still stable after wiring change

conclusions
interesting that delta/5" with the same idea 2 wiring as alpha 4.3" is more reliable
seems that it took 2 type a driving the hub in parallel WITH grounds connected
none of the old wiring "ideas" worked but strategy 3 seems to work ok
was the problem rooted in brownout or interference or both???
backlight flicker character has changed from intermittent to a constant high speed

final keyboard wiring
will certainly be more reliable to dedicate keyboard to type a port instead of wiring
wiring for keyboard could be a contributing factor to idea 2 wiring instability
given that delta/5" has similar wiring but no keyboard wiring and seems to work well

fate of delta/5"
add dedicated type a cable for keyboard wiring to eliminate any stability problems
not wired for keyboard currently anyway
is represenative of a failure to standardize devices

hub input wiring
second type a is wired in parallel
red wire is broken for switch
black wire and all others and metal ground connections are kept connected
additional shielding or wire twisting is not used
black wire is stripped but not broken for connection to second type a
hub direct to cellular and typical right/right angle microusb used for screen

Workaround:  

final results
delta/5"/340/2 type a with switch in parallel/no keyboard wiring - stable
alpha/4.3"/zte/2 type a with switch in parallel/keyboard wiring - unstable
alpha/4.3"/zte/hub with switch on input/no keyboard wiring - unstable
alpha/4.3"/zte/hub with 2 type a no ground on 2nd&switch/no keyboard wiring - unstable
alpha/4.3"/zte/hub with 2 type a switch on input/no keyboard wiring - stable

To summarize x86 MID usage - you may be able to use any wiring you want with the 5" screen.  If you use the 4.3" screen, then use a hub with 2 type a inputs bonded together for more power.  A switch can be attached to the power lines of each type A connector.  This switch will allow you to power down the attached components to save power after you put the x86 device to sleep.  You may be able to get away with using any wiring even on the 4.3" screen because this is highly dependent on the exact configuration of the device.

Models affected: All
